content/docs/examples/advanced-gateways/egress-mongo/index.md

```diff
@@ -0,0 +1,1111 @@
+    >}}
+
+Note that the MongoDB database is outside the Istio service mesh, or more precisely outside the Kubernetes cluster. The
+boundary of the service mesh is marked by a dashed line.
+
+### Access the webpage
+
+Access the webpage of the application, after
+[determining the ingress IP and port](/docs/examples/bookinfo/#determining-the-ingress-ip-and-port).
+
+Since you did not configure the egress traffic control yet, the access to the MongoDB service is blocked by Istio.
+This is why instead of the rating stars, the message _"Ratings service is currently unavailable"_ is currently
+ displayed below each review:
+
+{{< image width="80%" ratio="36.19%"
+    link="./errorFetchingBookRating.png"
+    caption="The Ratings service error messages"
+    >}}
+
+In the following sections you will configure egress access to the external MongoDB service, using different options for
+egress control in Istio.
+
+## Egress control for TCP
+
+Since [MongoDB Wire Protocol](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/) runs on top of TCP, you
+can control the egress traffic to your MongoDB as traffic to any other [external TCP service](/blog/2018/egress-tcp/). To
+control TCP traffic, a block of IPs in the [CIDR](https://tools.ietf.org/html/rfc2317) notation that includes the IP
+address of your MongoDB host must be specified. The caveat here is that the IP of the MongoDB host is not stable and
+could not be known in advance.
+
+In the cases when the IP of the MongoDB host is not stable, the egress traffic can either be
+[controlled as TLS traffic](#egress-control-for-tls), or the traffic can be routed
+[directly](/docs/tasks/traffic-management/egress/#calling-external-services-directly), bypassing the Istio sidecar
+proxies.
+
+Get the IP address of your MongoDB database instance. As an option, you can use the
+    [host](https://linux.die.net/man/1/host) command:
+
+{{< text bash >}}
+$ export MONGODB_IP=$(host $MONGODB_HOST | grep " has address " | cut -d" " -f4)
+{{< /text >}}
+
+### Control TCP egress traffic without a gateway
+
+In case you do not need to direct the traffic through an
+[egress gateway](/docs/examples/advanced-gateways/egress-gateway/#use-case), for example if you do not have a
+requirement that all the traffic that exists your mesh must exit through the gateway, follow the
+instructions in this section. Alternatively, if you do want to direct your traffic through an egress gateway, proceed to
+[Direct TCP egress traffic through an egress gateway](#direct-tcp-egress-traffic-through-an-egress-gateway).
+
+1.  Define a TCP mesh-external service entry:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: mongo
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      addresses:
+      - $MONGODB_IP/32
+      ports:
+      - number: $MONGODB_PORT
+        name: tcp
+        protocol: TCP
```

Comment:
I see that the various examples in this article use `protocol: TCP` or `protocol: TLS`. Given that `MONGO` is one of the [possible values](https://preliminary.istio.io/docs/reference/config/istio.networking.v1alpha3/#Port), shouldn't `protocol: MONGO` be used?

Reply:
I think it would be good to mention `protocol: MONGO` in the article and explain why you are using TCP instead, since readers would probably expect a ServiceEntry for mongodb would use it.

Reply:
Addressed in 9bac39be5c5fa5b85e322dc7bfe89e8431d16150.